shells/pipes-shell/web/cache-mgr.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+const ARCS_CACHE_MANAGER = './cache-mgr-sw.js';
+
+// Arcs Cache Manager
+(async function _arcsCacheManager() {
+  if ('serviceWorker' in navigator) {
+    // Service Worker supports ONLY https:// and http://localhost.
+    // Only https:// is accepted at this moment since local socket test
+    // servers are not supported on devices.
+    if (location.protocol !== 'https:') return;
+
+    // Only registering the new cache manager service worker instance when
+    // requested in 'https' protocol and with ?use-cache url parameter,
+    // otherwise unregistering the existing service workers at this scope.
+    const params = new URLSearchParams(window.location.search);
+    if (!params.has('use-cache')) {
+      navigator.serviceWorker.getRegistrations().then(function(regs) {
```

Comment:
You could use `await` here too, see comment below.

Reply:
ditto.